test/primitive_parser.js

```diff
@@ -19,6 +19,92 @@ describe('Primitive parser', function() {
       var buffer = Buffer.from([0x00, 0xd2, 0x04, 0x00, 0xbc, 0x61, 0x4e]);
       assert.deepEqual(parser.parse(buffer), { a: 0, b: 1234, c: 12345678 });
     });
+    describe.only('BigInt64 parsers', () => {
+      const [major] = process.version.replace('v', '').split('.');
+      if (Number(major) >= 12) {
+        it('should parse biguints64', () => {
+          const parser = Parser.start()
+            .biguint64be('a')
+            .biguint64le('b');
+          // from https://nodejs.org/api/buffer.html#buffer_buf_readbiguint64le_offset
+          const buf = Buffer.from([
+            0x00,
+            0x00,
+            0x00,
+            0x00,
+            0xff,
+            0xff,
+            0xff,
+            0xff,
+            0x00,
+            0x00,
+            0x00,
+            0x00,
+            0xff,
+            0xff,
+            0xff,
+            0xff,
+          ]);
+          assert.deepEqual(parser.parse(buf), {
+            a: 4294967295n,
```

Comment:
BigInt literal causes syntax error on earlier versions of node. We should use the BigInt(string) constructor.

Reply:
Ah good, point, this is a bit of a tricky one in terms of testing due to the nodejs requirements.
Thanks for pointing that out